internal/impl/kafka/franz_client.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"crypto/tls"
+	"strings"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/sasl"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+)
+
+const (
+	// Connection fields
+	kfcFieldSeedBrokers    = "seed_brokers"
+	kfcFieldClientID       = "client_id"
+	kfcFieldTLS            = "tls"
+	kfcFieldMetadataMaxAge = "metadata_max_age"
+)
+
+// FranzConnectionFields returns a slice of fields specifically for establishing
+// connections to kafka brokers via the franz-go library.
+func FranzConnectionFields() []*service.ConfigField {
+	return []*service.ConfigField{
+		service.NewStringListField(kfcFieldSeedBrokers).
+			Description("A list of broker addresses to connect to in order to establish connections. If an item of the list contains commas it will be expanded into multiple addresses.").
+			Example([]string{"localhost:9092"}).
+			Example([]string{"foo:9092", "bar:9092"}).
+			Example([]string{"foo:9092,bar:9092"}),
+		service.NewStringField(kfcFieldClientID).
+			Description("An identifier for the client connection.").
+			Default("benthos").
+			Advanced(),
+		service.NewTLSToggledField(kfcFieldTLS),
+		SASLFields(),
+		service.NewDurationField(kfcFieldMetadataMaxAge).
```

Comment:
Technically producers use metadata too. Metadata contains which brokers host which partitions, so if there are leadership changes or partitions move to partitions that are rarely written too it would prevent latency by discovering the change in the write. Most errors fallback to reading metadata and metadata refresh is mostly a tradeoff around extra traffic vs stale info. Anyways I doubt anyone would ever need to change this for an output practically.